common/src/main/java/io/netty/util/internal/PlatformDependent0.java

```diff
@@ -845,14 +847,38 @@ private static int javaVersion0() {
         if (isAndroid0()) {
             majorVersion = 6;
         } else {
-            majorVersion = majorVersionFromJavaSpecificationVersion();
+            majorVersion = majorVersionFromSystemClass();
         }
 
         logger.debug("Java version: {}", majorVersion);
 
         return majorVersion;
     }
 
+    // https://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html
+    private static int majorVersionFromSystemClass() {
+        InputStream in = ClassLoader.getSystemResourceAsStream("java/lang/ClassLoader.class");
```

Comment:
I'm slightly more worried that this can throw (like SecurityException, or other exceptions).  In that case, there does need to be a fallback.

Reply:
Yep